File: simple-stack-example-mvp/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/injection/SchedulerModule.java
package com.zhuinden.simplestackdemoexamplemvp.application.injection;

import com.zhuinden.simplestackdemoexamplemvp.util.SchedulerHolder;

import java.util.concurrent.Executors;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import rx.schedulers.Schedulers;

/**
 * Created by Owner on 2017. 01. 26..
 */

@Module
public class SchedulerModule {
    @Provides
    @Named("LOOPER_SCHEDULER")
    @Singleton
    SchedulerHolder looperScheduler(SchedulerHolder schedulerHolder) {
        return schedulerHolder;
    }

    @Provides
    @Named("WRITE_SCHEDULER")
    @Singleton
    SchedulerHolder writeScheduler(SchedulerHolder schedulerHolder) {
        schedulerHolder.setScheduler(Schedulers.from(Executors.newSingleThreadExecutor()));
        return schedulerHolder;
    }
}

File: samples/legacy-architecture-samples/simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/application/injection/SchedulerModule.kt
package com.zhuinden.simplestackdemoexamplemvp.application.injection

import com.zhuinden.simplestackdemoexamplemvp.util.SchedulerHolder
import dagger.Module
import dagger.Provides
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.Executors
import javax.inject.Named
import javax.inject.Singleton

/**
 * Created by Owner on 2017. 01. 26..
 */

@Module
class SchedulerModule {
    @Provides
    @Named("LOOPER_SCHEDULER")
    @Singleton
    fun looperScheduler(schedulerHolder: SchedulerHolder): SchedulerHolder = schedulerHolder

    @Provides
    @Named("WRITE_SCHEDULER")
    @Singleton
    fun writeScheduler(schedulerHolder: SchedulerHolder): SchedulerHolder = schedulerHolder.also { holder ->
        holder.scheduler = Schedulers.from(Executors.newSingleThreadExecutor())
    }
}
